cogs/monitor.py

```diff
@@ -13,96 +23,238 @@ async def on_message(self, message: discord.Message):
-    async def delete(self, ctx: commands.Context):
+    @message.command(aliases=['remove'], name='delete')
+    async def message_delete(self, ctx: commands.Context, *, pattern: RegexConverter):
         """
         Delete a monitor for a message pattern.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        monitored_message = await StaffMonitorMessage.filter(message=pattern).get_or_none()
+
+        if not monitored_message:
+            await ctx.send(f"The pattern (`{pattern}`) is not being monitored.")
+            return
+
+        await monitored_message.delete()
+        get_all_monitor_messages.cache_clear()
+
+        logging.debug(f"Removed pattern {pattern} from monitor.")
+        await ctx.send(f"The pattern (`{pattern}`) has been successfully removed from monitor.")
 
-    
     @monitor.group()
     async def user(self, ctx: commands.Context):
         """
         Commands for monitoring problematic users.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        await available_subcommands(ctx)
     
-    @user.command(aliases=['add'])
-    async def create(self, ctx: commands.Context):
+    @user.command(aliases=['add'], name='create')
+    async def user_create(self, ctx: commands.Context, member: discord.Member):
         """
         Create a monitor for a user.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        monitored_user = await StaffMonitorUser.filter(user_id=member.id).exists()
+
+        if monitored_user:
+            await ctx.send(f"{member.mention} is already being monitored.")
+            return
+
+        await StaffMonitorUser.create(user_id=member.id)
+        get_all_monitor_users.cache_clear()
+
+        logging.debug(f"Added user with id {member.id} to monitor.")
+        await ctx.send(f"{member.mention} has been successfully added to monitor.")
     
-    @user.command(aliases=['remove'])
-    async def delete(self, ctx: commands.Context):
+    @user.command(aliases=['remove'], name='delete')
+    async def user_delete(self, ctx: commands.Context, member: discord.Member):
         """
         Delete a monitor for a user.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        monitored_user = await StaffMonitorUser.filter(user_id=member.id).get_or_none()
+
+        if not monitored_user:
+            await ctx.send(f"{member.mention} is not being monitored.")
+            return
+
+        await monitored_user.delete()
+        get_all_monitor_users.cache_clear()
+
+        logging.debug(f"Removed user with id {member.id} from monitor.")
+        await ctx.send(f"{member.mention} has been successfully removed from monitor.")
+    
+    @monitor.command()
+    @admin_command()
+    async def toggle(self, ctx: commands.Context):
```

Comment:
I'm not sure if there is a use case for it, but we could maybe make it so that you can toggle on/off messages and user stuff separately?

Reply:
Not sure what the merit of this would be - this would also have to involve a migration of the DB schema